crates/dbsp/src/circuit/circuit_builder.rs

```diff
@@ -4580,7 +4580,8 @@ mod tests {
                 n += 1;
                 result
             }));
-            let (z1_output, z1_feedback) = circuit.add_feedback(Z1::new(0));
+            let (z1_output, z1_feedback) =
+                circuit.add_feedback(Z1::new(circuit.global_node_id().persistent_id(), 0));
```

Comment:
I think this returns the node id of the circuit itself, which is probably not what you want here?